addons/website_sale/views/templates.xml

```diff
@@ -816,25 +816,31 @@
                                     <select class="form-select css_attribute_select mb-2" name="attrib">
                                         <option value="" selected="true">-</option>
                                         <t t-foreach="a.value_ids" t-as="v">
-                                            <option t-att-value="'%s-%s' % (a.id,v.id)" t-esc="v.name" t-att-selected="v.id in attrib_set" />
+                                            <t t-if="v.is_product_variant_in_category(category) or v.id in attrib_set">
```

Comment:
What if instead of filtering on each value, we actually add a function to attributes to get all values available in the given category instead?
Otherwise we might also have cases where we display the title with no actual values that can be selected.